src/lesson10/library/Library.java

```diff
@@ -0,0 +1,75 @@
+package lesson10.library;
+
+import java.util.*;
+
+public class Library {
+    private List<Book> bookList = new ArrayList();
+    Comparator<Book> compareByTitle = (book1, book2) -> book1.getTitle().compareTo(book2.getTitle());
+    Comparator<Book> compareByDate = (book1, book2) -> -book1.getId().compareTo(book2.getId());
+
+    public Library() {
+    }
+
+    public List<Book> getBookList() {
+        return this.bookList;
+    }
+
+    public String stringBookList() {
+        String str = "";
+        Book book;
+        for (Iterator var2 = this.getBookList().iterator();
+             var2.hasNext(); str = str + "Title \"" + book.getTitle() + "\", genre \"" + book.getGenre() + "\"\n") {
+            book = (Book) var2.next();
+        }
+
+        return str;
+    }
+
+    public void setBookList(List<Book> bookList) {
+        this.bookList = bookList;
+    }
+
+    public void addBook(Book book) {
+        if (!this.getBookList().contains(book.getId())) {
+            this.bookList.add(book);
+        } else {
+            System.out.println("The book is already on the list!");
+        }
+
+    }
+
+    public void deleteBook(Integer id) {
+        this.bookList.removeIf((book) -> Objects.equals(id, book.getId()));
+    }
+
+    public Book get(Integer id) {
```

Comment:
don't like this method at all.
I'll write like this:
        for (Book book : this.bookList) {
            if (book.getId().equals(bookIdForSearch)) {
                return book;
            }
        }
        return null;
Of course streams are more efficient but you will learn them later
in this case it will be like this:
return this.bookList.stream().filter(book -> book.getId().equals(bookIdForSearch)).findFirst().orElse(null);

Reply:
fixed